src/main.rs

```diff
@@ -43,7 +43,19 @@ fn main() -> Result<(), Error> {
         for camera in config.cameras {
             s.spawn(move |_| {
                 // TODO handle these errors
-                let mut output = rtsp.add_stream(&camera.name).unwrap();
+                let mut format :&str = &camera.format;
+                if format == "h265" {
+                    format = "! h265parse ! rtph265pay name=pay0";
+                } else if format == "h264" {
+                    format = "! h264parse ! rtph264pay name=pay0";
+                } else {
+                    if format.trim().starts_with("!") {
+                        warn!("Using custom format: {}.", format);
+                    } else {
+                        error!("Unrecognised format: {}.", format);
+                    }
+                }
```

Comment:
Picking gstreamer launch arguments should be done in the `gst` module, not here, to keep the gstreamer wrangling encapsulated.

Create a new enum:
```rust
enum StreamFormat {
    H264,
    H265,
    Custom(String),
}
```
in the gst module and add it as an argument to the `add_stream` function. In that function, create a `match` block that generates the right gstreamer arguments. Then, here in main, write a `match` block that transforms this configuration string into the appropriate variant of that enum.

Reply:
(It will be slightly more lines of code my way, but it will also keep the API nice and clean.)

Reply:
Ok moved also added validation to the config toml file so it will error if they don't write a valid input e.g. h264. Learned lost of rust today :)